### ISSUE TITLE
no-shadow not work in double script block

### ISSUE THREAD

**Issue by teleskop150750** [vuejs/eslint-plugin-vue#2141]
#### Checklist

 I have tried restarting my IDE and the issue persists.
 I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.
Tell us about your environment

ESLint version: 8.34.0
eslint-plugin-vue version: 9.9.0
Node version: 18
Operating System:

#### Please show your full configuration:

```ts
/* eslint-env node */
require('@rushstack/eslint-patch/modern-module-resolution')

module.exports = {
  root: true,
  'extends': [
    'plugin:vue/vue3-essential',
    'eslint:recommended',
    '@vue/eslint-config-typescript',
    '@vue/eslint-config-prettier/skip-formatting'
  ],
  overrides: [
    {
      files: [
        'cypress/e2e/**/*.{cy,spec}.{js,ts,jsx,tsx}'
      ],
      'extends': [
        'plugin:cypress/recommended'
      ]
    }
  ],
  parserOptions: {
    ecmaVersion: 'latest'
  },
  rules: {
    'no-shadow': 'off',
    '@typescript-eslint/no-shadow': 'error',
  }
}
```
#### What did you do?

```vue
<script setup lang="ts">
const props = defineProps<{
  msg: string
}>()


function foo() {
  const props = 3

  return props
}
</script>

<script lang="ts">
export default {
  name: 'HelloWorld',
}
</script>

<template>
  <div class="greetings">
  </div>
</template>
```

#### What did you expect to happen?
Errors reported for the no-shadow rule.

#### What actually happened?
No errors reported for the no-shadow rule.

![image](https://user-images.githubusercontent.com/62594983/235674326-e050054f-b494-4609-8a81-7f1bc31c68c8.png)

![image](https://user-images.githubusercontent.com/62594983/235674577-4821ba49-6840-4b2d-a305-115dc73b6174.png)

**Comment by teleskop150750** [comment]
#### Repository to reproduce this issue
[https://github.com/teleskop150750/vue-no-shadow](https://github.com/teleskop150750/vue-no-shadow)

**Comment by FloEdelmann** [comment]
I think this is exactly the same issue as #1800. So I'll close this issue to keep the focus there. But thanks for opening the issue and providing a reproduction repository.

**Comment by teleskop150750** [comment]
> #### Репозиторий для воспроизведения этой проблемы
> [https://github.com/teleskop150750/vue-но-тени](https://github.com/teleskop150750/vue-no-shadow)

well now in vue 3.3 now defineOptions

**Comment by teleskop150750** [comment]
> Same problem
> 
> I wish someone would look into another issue with `no-shadow` that I reported over a year ago. #1800

well now in vue 3.3 now defineOptions
